app/services/reporting/DuplicateDetectionService.scala

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright 2017 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package services.reporting
+
+import model.persisted.{ UserApplicationProfile, UserIdWithEmail }
+import play.Logger
+import repositories.application.ReportingRepository
+import repositories.contactdetails.ContactDetailsRepository
+import repositories.faststreamContactDetailsRepository
+
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.Future
+
+case class DuplicateCandidate(email: String, firstName: String, lastName: String, latestProgressStatus: String)
+
+case class DuplicateApplicationGroup(matchType: Int, candidates: List[DuplicateCandidate])
+
+object DuplicateDetectionService extends DuplicateDetectionService {
+  val reportingRepository: ReportingRepository = repositories.reportingRepository
+  val cdRepository: ContactDetailsRepository = faststreamContactDetailsRepository
+}
+
+trait DuplicateDetectionService {
+  private val HighProbabilityMatchGroup = 1
+  private val MediumProbabilityMatchGroup = 2
+
+  val reportingRepository: ReportingRepository
+  val cdRepository: ContactDetailsRepository
+
+  def findAll: Future[List[DuplicateApplicationGroup]] = {
+    def toUserIdToEmailMap(cds: List[UserIdWithEmail]) = {
+      cds.groupBy(_.userId).mapValues(_.head.email)
+    }
+
+    for {
+      allCandidates <- reportingRepository.candidatesForDuplicateDetectionReport
+      candidatesEmails <- cdRepository.findEmails.map(toUserIdToEmailMap)
+    } yield {
+      val exportedApplications = allCandidates.filter(_.exportedToParity)
+      Logger.info(s"Detect duplications from ${allCandidates.length} candidates")
```

Comment:
This looks more like debug logging, is it from debug or did you intend this to be an info log?